tensorflow/core/kernels/conv_ops_gpu_3.cu.cc

```diff
@@ -413,6 +429,275 @@ struct PadInput<GPUDevice, T, int, NDIMS> {
   }
 };
 
+// TPF stands for Tile size Posibility Frontier. It denotes the tile size
+// combinations that consume the most computational resources constrained by
+// number of threads per SM limit, shared memory limit and overall degree of
+// parallelism.
+// Tile size combinations lying on this frontier would achieve the maximum
+// utilization of available resources and combinations lying outside this
+// frontier is not performance-wise profitable or simply not possible.
+#define TPF_CHECKER(OP)                                                   \
+  size_of_t == 16 &&    (TileLongSide == 32   && (TileShortSide OP 4)  || \
```

Comment:
You went through all this work to indent this macro nicely, but then this first line is odd.  :)